db/users.go

```diff
@@ -26,23 +26,25 @@ type PrincipalConfig struct {
 // Check if the password in this PrincipalConfig is valid.  Only allow
 // empty passwords if allowEmptyPass is true.
 func (p PrincipalConfig) IsPasswordValid(allowEmptyPass bool) (isValid bool, reason string) {
-
 	// if it's an anon user, they should not have a password
-	if p.Name == nil && p.Password != nil {
-		return false, "Anonymous users should not have a password"
+	if p.Name == nil {
+		if p.Password != nil {
+			return false, "Anonymous users should not have a password"
+		} else {
+			return true, ""
+		}
 	}
 
 	if allowEmptyPass {
```

Comment:
Should this allowEmptyPass only be returning true if password==nil?  Otherwise it seems like we're allowing invalid (one character) passwords on an update in GetPrincipal (when allowReplace is true).